src/foam/dao/index/TreeNode.java

```diff
@@ -345,4 +357,24 @@ public TreeNode lte(TreeNode s, Object key, PropertyInfo prop) {
       s.level, s.left, null);
   }
 
+  public Object[] getAllValues(Object s){
+    Object[] nodeList = new Object[(int)((TreeNode)s).size];
+    getValuesHelper((TreeNode)s,nodeList,0);
+    return nodeList;
+  }
+  public int getValuesHelper(TreeNode currentNode, Object[] nodeList, int i){
+    if(currentNode.getValue()!=null)
+      nodeList[i]=(currentNode.getValue());
+    i++;
+    TreeNode left = currentNode.getLeft();
```

Comment:
Process left branch before processing getValue() so you get an in-order traversal (ie. sorted correctly).